WordPress/src/main/java/org/wordpress/android/models/Note.java

```diff
@@ -482,6 +482,21 @@ public JSONArray getHeader() {
         }
     }
 
+    // this method is used to compare two Notes: as it's potentially a very processing intensive operation,
+    // we're only comparing the note id, timestamp, and raw JSON length, which is accurate enough for
+    // the purpose of checking if the local Note is any different from a remote note.
+    public boolean similar(Note note) {
```

Comment:
Good observation! I actually started by overriding the `equals` method on the `Note` class but then realized that in order to actually implement `equals` meaningfully we'd have to thoroughly check the JSON objects representing the Notes (as coming from the server) should be deeply compared. So that's why I chose to be a bit lazier and so called this `similar` rather than `equal`.
That said, I think `equalsTimeAndLength` is way clearer! I'll address that.